### ISSUE TITLE
recommended way to `cabal install --global`?

### ISSUE THREAD

**Issue by jwaldmann** [haskell/cabal#7172]
I am the instructor for some programming-related classes, and I want to install (compiler, tools, and) libraries for all students to use, in a computer pool (currently, for remote use only, but that does not matter much here).

Yes yes at some point students will have to learn how to do that on their own (and trust binary downloads ...) but in the beginning of courses, this detracts from actual teaching. I rather show how to use `ssh` (possibly `x2go`), and be done.

So, I was used to `cabal install --global` the packages that I need.   The following worked until recently,
but for `cabal-install-3.2.0.0` and `ghc-8.10.2` it does not, and I find it hard to trace where it's going wrong.

Ghc is configured with  `--prefix=/opt/ghc/ghc-8.10.2`,
students put `/opt/ghc/ghc-8.10.2/bin` in their PATH.

In my  `.cabal/config`:
```
install-dirs global
  prefix: /opt/ghc/$compiler
```
Then (after checking that `--user` install works), e.g.,
```
cabal install --global --lib csound-expression  
```
With 8.10.2, this still puts in my `HOME/.ghc/x86_64-linux-8.10.2/environments/default`
```
package-id csound-expression-5.3.3-dc2...
```
and when I then start `ghci` (no arguments), I get
```
<command line>: cannot satisfy -package-id csound-expression-5.3.3-dc2...
```

NB: Yes, I guess it's fighting windmills (also sometimes called "progress") - I guess nobody really has this problem as they all install a complete binary distribution of compilers, tools and libraries in their HOME (add a bloated Editor  for good measure)  but  that seems such a waste. Or - expose students to surveillance by "online programming" web sites and their 3rd-party JS ...

Just sayin - `cabal` did work for my use case - I wonder what I am missing now.

**Comment by fgaz** [comment]
I'm guessing it worked with cabal-install <3.0, since that's when cabal-install switched to [a nix-like model](https://cabal.readthedocs.io/en/3.4/nix-local-build-overview.html).

Related: #6515, the solution in there may work for you too.

> `<command line>: cannot satisfy -package-id csound-expression-5.3.3-dc2...`

I'm not sure what's going on here. `--global` shouldn't even do anything with v2- commands. Maybe the environment file got broken somehow.

**Comment by fgaz** [comment]
> after checking that --user install works

aha. `cabal install --lib` has some problems with subsequent invocations at the moment. if you want to use this method to install libraries, try deleting the environment file, then install them all at once (`cabal install --lib pkg1 pkg2 pkg3...`)

**Comment by phadej** [comment]
Let us not recommend `cabal install --lib` until it works (i.e. someone fixes it).

**Comment by jwaldmann** [comment]
> Let us not recommend `cabal install --lib` until it works (i.e. someone fixes it).

But `cabal` *is* recommending it:
```
cabal install csound-catalog
Resolving dependencies...
...
Warning: You asked to install executables, but there are no executables in
target: csound-catalog. Perhaps you want to use --lib to install libraries
instead.

cabal-install version 3.2.0.0
compiled using version 3.2.1.0 of the Cabal library 
```

**Comment by Mikolaj** [comment]
@singpolyma: yes, it is (ot just not having particularly useful functionality for most users). Please kindly try the several other approaches mentioned in #6481 and report there if any fits your use case.

**Comment by singpolyma** [comment]
My current use case is that I have a git clone, and I try to install it with `cabal install --lib` and then cd to the directory of a project that has this one as a dependency and run `cabal configure --enable-tests` over there, but it fails to find the package I installed in the first step.  With `v1-install` it all works.

**Comment by jneira** [comment]
Oh, for that use case there is a more direct solution as you are gonna use a project anyways: you can create a cabal.project in the second project (or update the existing one) with

```
packages: *.cabal /path/to/git/clone
```

(assuming the cabal project hasonly one .cabal file)
